app/controllers/checkout_controller.rb

```diff
@@ -62,6 +68,22 @@ def update
 
   private
 
+  # Check if items present in cart have been removed from the OC
+  # before checkout has been completed
+  def recheck_stock_availability
+    outgoing_variants = @order.order_cycle.exchanges.outgoing.first.variants.pluck(:id)
+    items_available = true
```

Comment:
you don't need to initialize any variable if you use [each_with_object](https://apidock.com/rails/Enumerable/each_with_object) instead of each.